packages/url_launcher/url_launcher/lib/src/link.dart

```diff
@@ -90,16 +90,17 @@ class DefaultLinkDelegate extends StatelessWidget {
   bool get _useWebView {
     if (link.target == LinkTarget.self) return true;
     if (link.target == LinkTarget.blank) return false;
-    return null;
+    return false;
   }
 
   Future<void> _followLink(BuildContext context) async {
-    if (!link.uri.hasScheme) {
+    if (link.uri == null || !link.uri!.hasScheme) {
       // A uri that doesn't have a scheme is an internal route name. In this
       // case, we push it via Flutter's navigation system instead of letting the
       // browser handle it.
       final String routeName = link.uri.toString();
-      return pushRouteNameToFramework(context, routeName);
+      await pushRouteNameToFramework(context, routeName);
+      return;
```

Comment:
Is this nnbd or just better practice? If pushRouteNameToFramework returns Future<void>, why not return that from here? Is it because it's in another package and it's some `Future<void>?`?

Reply:
I'm wondering the same thing.

Reply:
Otherwise, the analyzer complains:
```
A value of type 'ByteData' can't be returned from method '_followLink' because it has a return type of 'Future<void>'.dartreturn_of_invalid_type
```

Also, it doesn't seem like the intention is to return `Future<ByteData>`.

Reply:
Ooooh good catch by the analyzer, I wonder how come that wasn't an error before nnbd!

Reply:
Casting anything to `void` used to be fine. I'm not sure why it changed with nnbd.